public/app/core/utils/richHistory.test.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  addToRichHistory,
+  updateStarredInRichHistory,
+  updateCommentInRichHistory,
+  mapNumbertoTimeInSlider,
+  createDateStringFromTs,
+  createQueryHeading,
+  createDataQuery,
+} from './richHistory';
+
+import { SortOrder } from './explore';
+
+describe('addToRichHistory', () => {
```

Comment:
Not sure about these tests here. The main point of these functions is the side effect of saving stuff inside the localstorage. So I think it would be better to mock out localstorage and test if it was saved there. Not sure if also use the return value, if we do than we should keep the test for that also but that is only a part of the funcitonality. Also if we already added the error handling testing for error conditions would also be nice.

Reply:
Tested. Error handling is still not part of Rich history, so I will include tests in that PR.